solutions/usaco-895.mdx

```diff
@@ -2,16 +2,96 @@
 id: usaco-895
 source: USACO Silver 2019 January
 title: Icy Perimeter
-author: Tanish Tyagi
+author: Tanish Tyagi, Brad Ma
 ---
 
-[Official Analysis](http://www.usaco.org/current/data/sol_perimeter_silver_jan19.html)
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_perimeter_silver_jan19.html)
+
+## Implementation
+**Time Complexity:** $\mathcal{O}(N^2)$
 
 <LanguageSection>
+	
+<JavaSection>
+	
+```java
+import java.io.*;
+import java.util.*;
 
-<CPPSection>
+public class IcyPerimeter {
```

Comment:
```suggestion
public class Perimeter {
```
so it matches the actual problem name

Reply:
the actual problem name is icy perimeter